starwars_api/models.py

```diff
@@ -55,27 +63,64 @@ def __repr__(self):
 class BaseQuerySet(object):
 
     def __init__(self):
-        pass
+        self.next_page = None
+        self.previous_page = None
+        self.results = []
+        self.current_record = 0
+        self.current_page = 0
 
     def __iter__(self):
-        pass
+        return self
 
     def __next__(self):
         """
         Must handle requests to next pages in SWAPI when objects in the current
         page were all consumed.
         """
-        pass
-
+        while True:
```

Comment:
i assume that even though it's never set to false, it won't be an infinite loop because you'll eventually hit `StopIteration` that'll break the loop. am I on the right track?